### DISCUSSION TITLE
Jikong BMS RS485 Module and Victron CCGX

### DISCUSSION THREAD

**Discussion by ghost** [Louisvdw/dbus-serialbattery#147] [General]
Hi Louis! First of all thank you for all this great work! I have been reading along for quite some time, but have not been ready to try out your serial battery driver until now.

Background:
I am currently aboard a sailboat that our Canadian family has been sailing around the world for the past 7 years starting in Turkey and we are currently in SE Asia with intentions of sailing to South Africa next year - so hopefully I can buy you a couple of drinks in the future personally. I really like the fact that the driver allows me to see the individual batteries during inland travel away from the boat.

Current Install:
I have recently built 4 x 12v batteries comprised of 304Ah EVE cells (~1200Ah) contained in prefab plastic cases (tight fit) each with a JK BMS installed and they have been running well for the past few weeks after the top balancing process was completed. They are each individually fused with BlueSeas terminal fuses on the positive terminals, as well as individual shut-off switches before being combined into a Lynx "Power in" distributor which includes an additional "Internal multi fuse hack" then out through a large 600A T-fuse (+ve) and Victron BMV712 shunt (-ve) before tying into another Lynx distributor module connecting all our systems onboard which include a Multiplus 3000W inverter and slightly older model Blue Solar MPPT 150/85 which does not use Bluetooth, but is connected via Can-bus to our Multiplus with the latest firmware updates. Everything is visible on our existing VRM through our Colour Control GX, with current firmware version 2.87.

I have purchased both the expensive 4 x Victron RS485 to USB cables and the required 4 x Jikong RS485 Serial modules and researched their pin outs and see that:

i) the Victron cable, Yellow is B (Data -), Orange is A (Data +), Black in Ground.  There are 3 additional wires including RED for power.
ii) on the Jikong Module, White is B, Yellow is A, Black is Ground.

Questions:
1. I intend to connect just these 3 corresponding wires. Would anyone please confirm whether I also need to connect the RED power wire somehow and to what? I understand the Victron cable is isolated power wise, but I do not understand how that comes into play with the JK BMS, so some guidance there would be appreciated.

2. I intend to combine all the Victron USB Cables to a USB hub, then connect the hub directly to the Colour Control GX. Is there any issues with this proposed installation?

3. With the Colour Control GX, I BELIEVE my access to the OS is quite limited. I understand from the wiki that I may install the Serial Battery Driver using a memory card, but am concerned about uninstallation if there is an issue. The wiki mentions using SSH for removal, but I am not aware of that functionality on the CCGX....So I have a concern, that I would appreciate some clarification.

4. Do you have any concerns regarding the version of my CCGX before I attempt to install the driver?

Idea:
As the JK BMS has the ability to manually turn-off charging and discharging, I believe it would be an added value to be able to remotely control these aspects through the VRM, if I were to see a large battery disparity happening while away from the boat. I would appreciate your thoughts on this...

Kind regards and thank you in advance,

Shaun
s/y Element (currently moored in Phuket, Thailand)

**Comment by Louisvdw** [comment]
Hi Shaun

That sounds like a very nice trip! Come say hi when you are in SA. :) 

Even though your Color Control GX is one of the slowest GX devices, it will all work 100% on the CCGX. The driver will work on any GX device (including the CCGX) that can run the GX firmware from around V2.60 and newer. A USB-hub is also no issue (I also use one).

If your CCGX is connected on a network (sounds like it if you want to monitor it while away from the boat) then you can connect to it using a PC on the same network and any SSH client (Kitty/Putty on Windows or ssh on MacOS/Linux). You enable this access on the Remote Console settings. If you have SSH access then you can disable the driver at any time. 
The other option would be to switch your Battery Monitor to your BMV or Multiplus (it will be on the driver as default after it is installed) and it will use that value and not the BMS and your BMS will be just for information.

For the connection of the BMS using the Victron isolated cable you will need to connect the red power wire as well, but on the JKBMS this pin for the RS485 does not give you 5V, but instead the battery voltage. You will need to connect it to a 5V pin on the BMS if you have a multimeter that you can test the pins with. If you have the JKBMS RS485 cable then they it already convert this for 5V and you can just use that. 

The last thing too remember is that you will currently see the detail for all the batteries, but only the value for 1 will be on the main display. Ticket #8 will solve that.
